packages/react-instantsearch-hooks/src/lib/useWidget.ts

```diff
@@ -31,13 +32,16 @@ export function useWidget<TWidget extends Widget | IndexWidget, TProps>({
   const shouldAddWidgetEarly =
     shouldSsr && !parentIndex.getWidgets().includes(widget);
 
+  const search = useInstantSearchContext();
+
   // This effect is responsible for adding, removing, and updating the widget.
   // We need to support scenarios where the widget is remounted quickly, like in
   // Strict Mode, so that we don't lose its state, and therefore that we don't
   // break routing.
   useIsomorphicLayoutEffect(() => {
     const previousWidget = prevWidgetRef.current;
     function cleanup() {
+      if (search._preventWidgetCleanup) return;
```

Comment:
Strictly speaking, we probably need to have `search` as a dependency of the effect now. You could temporarily change `useIsomorphicLayoutEffect()` to `useEffect()` to respect the ESLint rule.

Reply:
Good catch! I customized the ESLint rule to include the custom hook

```js
'react-hooks/exhaustive-deps': [
  'warn',
  {
    additionalHooks: '(useIsomorphicLayoutEffect)',
  },
],
```

Now it's warning about `props`, should we add it to the array of deps?

Reply:
I think we should add `props` to the dependency array as well, we expect to delete/add the widget when only the props change.